csharp.test/TestLogicalTypeRoundtrip.cs

```diff
@@ -143,6 +143,68 @@ public static void TestDateTimeRoundTrip(DateTimeKind kind, TimeUnit timeUnit)
             Assert.AreEqual(kind, kinds.First());
         }
 
+        [Test]
+        [NonParallelizable]
+        public static void TestAppSwitchDateTimeKindUnspecified()
+        {
+            AppContext.SetSwitch("ParquetSharp.UseDateTimeKindUnspecified", true);
```

Comment:
I think that it would be good to assert that the switch is not set before we start manipulating it. The precondition of this test that all other tests expect the switch to be false. In case someone uses registry entry to set a switch he would get weird tests errors, so such assertion would help to debug such issue.